### PR TITLE
Jormungandr : Use gevent patch ssl

### PR DESCRIPTION
**Pull request by mbounabi**
https://jira.canaltp.fr/browse/NAVP-779

Résultats:
https://jira.canaltp.fr/browse/NAVP-780
https://jira.canaltp.fr/browse/NAVP-781
https://jira.canaltp.fr/browse/NAVP-782

### CODE REVIEW

**Review by xlqian — APPROVED** [review]
We may want to enable patch_ssl and patch_socket in the integration test https://github.com/CanalTP/navitia/blob/dev/source/jormungandr/jormungandr/test_settings.py @kinnou02 @pbougue @TeXitoi

File: source/jormungandr/jormungandr/api.py [hunk 1]
```diff
@@ -48,15 +48,18 @@
...
-    # the reason why we don't use patch_socket() at the very beginning of jormungandr is
-    # that it caused a mysterious performance regression for certain requests, thus we patch
-    # only at places where asynchronisation is needed
     # Note that "monkey_patch" only patches on http request because we want asynchronisation on that,
     # but we don't want that for reddis because it may cause performance regression
     import requests
     import gevent.socket
     requests.packages.urllib3.connection.connection.socket = gevent.socket
 
+if rest_api.app.config.get('PATCH_GEVENT_SSL', False):
```
Comment (kinnou02) [inline source/jormungandr/jormungandr/api.py]: I don't see a case where we want to patch the socket but not the ssl since it will break things

Comment (mbounabi) [reply source/jormungandr/jormungandr/api.py]: Not pertinent indeed. Changed!

**Review by kinnou02 — APPROVED** [review]

**Comment by TeXitoi** [comment]
@xlqian we can but it will not test so much as we don't call http services in the tests.

**Review by kinnou02 — APPROVED** [review]

**Comment by xlqian** [comment]
@TeXitoi yes, i'm aware of that, I mean, just in case those patches have some side effects(that we haven't detected yet) on the code, the integration tests will **_roughly_** protect us

**Comment by kinnou02** [comment]
You can change it for the tests and also for every environement except the production (You can also create the PR for prod already but with donotmerge tag), it will help us see if there is any problems.

**Comment by mbounabi** [comment]
jenkins, test this please!

**Review by benoit-bst — APPROVED** [review]

**Review by pbougue — APPROVED** [review]
OK for me.
As Alex said, yo can open a PR on deployment conf, co that every platform is like dev, except for prod : https://github.com/CanalTP/navitia_deployment_conf/blob/master/dev.py#L187
And also a "do_not_merge"-PR for prod aside.

**Comment by mbounabi** [comment]
PRs on navitia_deployment_conf have been created:
https://github.com/CanalTP/navitia_deployment_conf/pull/368
https://github.com/CanalTP/navitia_deployment_conf/pull/369

**Pull request merged**